Revert Liquibase to v4.20.0

### Describe the bug

See my original comment: https://github.com/quarkusio/quarkus/commit/d05e504f9af2532ded814ee4e3b4a44d81b616ac#r117808146

Liquibase v4.21.0 and v4.21.1 contains some bugs. One of them is critical: it's a bug on the hash calculations when creating procedures in PostgreSQL. This prevent Liquibase to apply the changes, and so Quarkus fails to start. [The recommended solution is to downgrade to v4.20.0](https://github.com/liquibase/liquibase/issues/4156#issuecomment-1559473901) (do not use v4.22.0, it was even worse and they removed it from the repos).

Here's an output of Quarkus starting since I upgraded from 3.0.4 to 3.1.1:
```
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:273)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:250)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:592)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:640)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	... 37 more
Caused by: java.lang.IllegalStateException: Error starting Liquibase
	at io.quarkus.liquibase.runtime.LiquibaseRecorder.doStartActions(LiquibaseRecorder.java:93)
	at io.quarkus.deployment.steps.LiquibaseProcessor$startLiquibase1744275855.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LiquibaseProcessor$startLiquibase1744275855.deploy(Unknown Source)
	... 53 more
Caused by: liquibase.exception.ValidationFailedException: Validation Failed:
     3 changesets check sum
          db/changelogs/2023-01-11_202_foo.xml::202-foo:add-timestamp-function::X was: 8:bcb966fbe934329da77f2c259a3f8adf but is now: 8:e350d4815158265f13ef42a2b920c97a
          db/changelogs/2023-02-03_227_bar.xml::227-bar:add-procedures::X was: 8:80cacbfb7672ee1e73e11ba890b749f0 but is now: 8:a3425cadf69ee4be00863edda2ddc65a
          db/changelogs/2023-03-02_248_baz.xml::248-baz:remove-trigger::X was: 8:d4e0e8065bf4e43cc59cfc50e820a8e0 but is now: 8:f259a5df7b4dd551f7b1e925221076c1

	at liquibase.changelog.DatabaseChangeLog.validate(DatabaseChangeLog.java:357)
	at liquibase.changelog.DatabaseChangeLog.validate(DatabaseChangeLog.java:330)
	at liquibase.Liquibase.validate(Liquibase.java:2577)
	at io.quarkus.liquibase.runtime.LiquibaseRecorder.doStartActions(LiquibaseRecorder.java:80)
	... 55 more
```

### Expected behavior

Freeze Liquibase v4.20.0 until a new version is out fixing all these bugs, and so do not prevent Quarkus to start.

### Actual behavior

Quarkus will not start if your project:
- Uses Quarkus v3.1.0 or v3.1.1
- Contains `quarkus-liquibase` dependency
- One of the Liquibase changeset contains `createProcedure`

See https://github.com/liquibase/liquibase/issues/4156

### How to Reproduce?

TODO: create a project to reproduce

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -168,7 +168,7 @@
         <jboss-logmanager.version>1.1.1</jboss-logmanager.version>
         <flyway.version>9.17.0</flyway.version>
         <yasson.version>3.0.3</yasson.version>
-        <liquibase.version>4.21.0</liquibase.version>
+        <liquibase.version>4.20.0</liquibase.version>
         <snakeyaml.version>2.0</snakeyaml.version>
         <osgi.version>6.0.0</osgi.version>
         <mongo-client.version>4.9.1</mongo-client.version>

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -168,9 +168,7 @@
         <jboss-logmanager.version>1.1.1</jboss-logmanager.version>
         <flyway.version>9.17.0</flyway.version>
         <yasson.version>3.0.3</yasson.version>
-        <!-- these two artifacts needs to be compatible together -->
-        <liquibase.version>4.21.1</liquibase.version>
-        <liquibase-mongodb.version>4.21.0</liquibase-mongodb.version>
+        <liquibase.version>4.21.0</liquibase.version>
         <snakeyaml.version>2.0</snakeyaml.version>
         <osgi.version>6.0.0</osgi.version>
         <mongo-client.version>4.9.1</mongo-client.version>
@@ -6167,7 +6165,7 @@
             <dependency>
                 <groupId>org.liquibase.ext</groupId>
                 <artifactId>liquibase-mongodb</artifactId>
-                <version>${liquibase-mongodb.version}</version>
+                <version>${liquibase.version}</version>
                 <exclusions>
                     <exclusion>
                         <groupId>org.codehaus.groovy</groupId>

